### PR TITLE
line protocol encoder

### PR DESCRIPTION
**Pull request by docmerlin**
An encoder for the line protocol

### CODE REVIEW

File: line-protocol/metric.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+// Field holds the keys and values for a bunch of Metric Field k/v pairs
+type Field struct {
+	Key   string
+	Value interface{}
+}
+
+// Metric is the interface for marshaling, if you implement this interface you can be marshalled into the line protocol.  Woot!
+type Metric interface {
```
Comment (desa) [inline line-protocol/metric.go]: Would be great if I didn't have to implement this interface to be able to write metrics.

Comment (desa) [reply line-protocol/metric.go]: Maybe something using struct tags?

Comment (docmerlin) [reply line-protocol/metric.go]: We were trying to avoid having to do a lot of reflection, which is why we went with this.

Comment (docmerlin) [reply line-protocol/metric.go]: Here's the original PR.
https://github.com/influxdata/line-protocol/pull/1

File: line-protocol/metric.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// Tag holds the keys and values for a bunch of Tag k/v pairs
+type Tag struct {
+	Key   string
+	Value string
+}
+
+// Field holds the keys and values for a bunch of Metric Field k/v pairs
+type Field struct {
+	Key   string
+	Value interface{}
```
[the remark this answers is not in the document]
Comment (docmerlin) [reply line-protocol/metric.go]: It would be pretty easy to make one as a wrapper around the Encoder.

**Comment by docmerlin** [comment]
Going to close this.

**Pull request closed without merging**